### PR TITLE
Aladin#getViewImageBuffer support

### PR DESCRIPTION
**Pull request by Xen0Xys**
For testing purposes, it il possible to save a PNG ArrayBuffer with the following code :
```js
const save = function(arrayBuffer, filename) {
    const blob = new Blob([arrayBuffer], { type: 'image/png' });
    const url = URL.createObjectURL(blob);
    const a = document.createElement('a');
    a.href = url;
    a.download = filename;
    document.body.appendChild(a);
    a.click();
    document.body.removeChild(a);
    URL.revokeObjectURL(url);
}
```

### CODE REVIEW

File: src/js/View.js [hunk 1]
```diff
@@ -446,42 +446,65 @@ export let View = (function () {
         this.catalogCanvas.style.cursor = cursor;
     };
 
-    /**
-     * return dataURL string corresponding to the current view
-     */
-    View.prototype.getCanvasDataURL = async function (imgType, width, height) {
+    View.prototype.getCanvasContext = async function (imgType, width, height, withLogo=true) {
```
Comment (bmatthieu3) [inline src/js/View.js:449]: I would better call it getCanvas

File: src/js/Aladin.js [hunk 2]
```diff
@@ -2172,11 +2174,16 @@ aladin.on("positionChanged", ({ra, dec}) => {
...
             options.format,
             options.width,
-            options.height
+            options.height,
+            options.logo
         );
         return canvasDataURL;
     };
 
+    Aladin.prototype.getViewImageBuffer = async function (withLogo) {
```
Comment (bmatthieu3) [inline src/js/Aladin.js:2183]: It is returning an array buffer so I would call it getViewArrayBuffer

File: src/js/Aladin.js [hunk 3]
```diff
@@ -2172,11 +2174,16 @@ aladin.on("positionChanged", ({ra, dec}) => {
...
+        return await this.view.getCanvasBuffer("image/png", null, null, withLogo);
```
Comment (bmatthieu3) [inline src/js/Aladin.js:2184]: I would call it getCanvasArrayBuffer too

**Comment by bmatthieu3** [comment]
Maybe also I think it would be good to add some jsdoc for Aladin.prototype.getViewDataURL and Aladin.prototype.getViewArrayBuffer.
Even maybe do a new method call Aladin.prototype.getView(dataType) with datatype that can be: 'blob', 'url', 'arrayBuffer'

**Comment by Xen0Xys** [comment]
> Maybe also I think it would be good to add some jsdoc for Aladin.prototype.getViewDataURL and Aladin.prototype.getViewArrayBuffer. Even maybe do a new method call Aladin.prototype.getView(dataType) with datatype that can be: 'blob', 'url', 'arrayBuffer'

Do you mean a function to replace `getCanvasArrayBuffer` and `getCanvasDataURL`, or a function that calls these two?

**Pull request merged**